### ISSUE TITLE
schema: draining name stays forever in namespace and table descriptor when job lost

### ISSUE THREAD

**Issue by itsbilal** [cockroachdb/cockroach#44253]
When an `ALTER TABLE RENAME` is run and its transaction is committed, the table descriptor on the current node is updated with the new table name. However, the old table name sticks around in the namespace and in the descriptor's `DrainingNames` field, until a job enqueued by the schema changer runs non-transactionally in the future when there are no more outstanding references to the old table name. The docs on `ALTER TABLE RENAME` mention some limitations of this mechanism.

It's possible (due to network issues, node restarts, etc) to permanently have an orphaned table name sticking around in `DrainingNames` if the job payload does not get persisted in the job table at the same time the `ALTER TABLE RENAME` commits.

A customer experienced this in production, and their logs have repeated instances of these lines:

```
I200120 11:28:28.844347 146 sql/lease.go:352  [n1,scExec] publish: descID=193 (balance_reserved_histo
ry_old) version=20 mtime=2020-01-20 11:28:28.843598733 +0000 UTC
W200120 11:28:28.850972 146 sql/schema_changer.go:1586  [n1] Error executing schema change: job with
ID 494863007173640194 does not exist
```
The job ID printed out is always the same, highlighting an orphaned job reference in the table descriptor.

This Zendesk ticket (requires login) contains more contextual info as well as a debug zip about the issue observed by the customer: https://cockroachdb.zendesk.com/agent/tickets/4474 , however, in the absence of logs going back a sufficient amount of time, it's hard to pinpoint to what caused the job to not be persisted in the first place. Note that the original `ALTER TABLE RENAME` is also missing from `events.json`, which probably means there was some unknown transient issue in the cluster that allowed the descriptor to be updated but didn't lead to the events/jobs table getting updated.

**Comment by itsbilal** [comment]
Would probably be a good idea if someone more familiar with schema changes triaged this first.

/cc @lucy-zhang in case you wanna triage this or assign this.

**Comment by pbardea** [comment]
Note: this issue was seen on v19.1.5.

Digging into this issue and briefly synced with @lucy-zhang, this issue should be resolved as of 19.2 (and should also not be present in 20.1).
